packages/zero-client/src/client/zero.ts

```diff
@@ -477,7 +478,9 @@ export class Zero<
             mutatorOrMutators,
             schema,
             slowMaterializeThreshold,
-          );
+            // Replicache expects mutators to only be able to return JSON
+            // but Zero wraps the return with: `{server?: Promise<MutationResult>, client?: T}`
+          ) as () => MutatorReturn;
```

Comment:
this is unfortunate. We could widen the replicache type.

Reply:
Yes. I don't think we serialize the result to JSON anywhere so this  type restriction was not necessary.